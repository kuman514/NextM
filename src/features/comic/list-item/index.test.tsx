import { render, screen } from '@testing-library/react';

import ComicListItem from '.';

jest.mock(
  'next/link',
  () =>
    function MockedLink(props: React.AnchorHTMLAttributes<HTMLAnchorElement>) {
      return <a {...props} />;
    }
);

describe('ComicListItem', () => {
  it('should render as expected (current: thumbnail, title, description)', () => {
    const { container } = render(
      <ComicListItem
        comicData={{
          id: 1234567,
          digitalId: 0,
          issueNumber: 0,
          variantDescription: '',
          isbn: '',
          upc: '',
          title: 'Koishi Komeiji',
          description: '',
          diamondCode: '',
          ean: '',
          issn: '',
          format: '',
          pageCount: 0,
          textObjects: [],
          images: [],
          modified: new Date(),
          thumbnail: {
            path: 'http://i.annihil.us/u/prod/marvel/i/mg/c/e0/535fecbbb9784',
            extension: 'jpg',
          },
          resourceURI: 'http://gateway.marvel.com/v1/public/characters/1011334',
          characters: {
            available: 0,
            collectionURI: '',
            items: [],
            returned: 0,
          },
          stories: {
            available: 0,
            collectionURI: '',
            items: [],
            returned: 0,
          },
          series: {
            resourceURI: '',
            name: '',
          },
          urls: [],
        }}
      />
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should has href=/comic/1234 when the comicData is having id=1234', async () => {
    render(
      <ComicListItem
        comicData={{
          id: 1234,
          digitalId: 0,
          issueNumber: 0,
          variantDescription: '',
          isbn: '',
          upc: '',
          title: 'Koishi Komeiji',
          description: '',
          diamondCode: '',
          ean: '',
          issn: '',
          format: '',
          pageCount: 0,
          textObjects: [],
          images: [],
          modified: new Date(),
          thumbnail: {
            path: 'http://i.annihil.us/u/prod/marvel/i/mg/c/e0/535fecbbb9784',
            extension: 'jpg',
          },
          resourceURI: 'http://gateway.marvel.com/v1/public/characters/1011334',
          characters: {
            available: 0,
            collectionURI: '',
            items: [],
            returned: 0,
          },
          stories: {
            available: 0,
            collectionURI: '',
            items: [],
            returned: 0,
          },
          series: {
            resourceURI: '',
            name: '',
          },
          urls: [],
        }}
      />
    );

    const link = await screen.findByLabelText('list-item-link');
    expect((link as HTMLAnchorElement).href).toStrictEqual(
      'http://localhost/comic/1234'
    );
  });
});
