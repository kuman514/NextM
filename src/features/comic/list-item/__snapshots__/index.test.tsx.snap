// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ComicListItem should render as expected (current: thumbnail, title, description) 1`] = `
<li>
  <a
    aria-label="list-item-link"
    class="flex flex-col w-48 h-72 rounded-xl overflow-hidden bg-black hover:bg-red-700 transition"
    href="/comic/1234567"
  >
    <div
      class="w-48 h-48"
    >
      <img
        alt="Koishi Komeiji"
        class="w-full h-full object-cover"
        data-nimg="1"
        decoding="async"
        height="192"
        loading="lazy"
        src="/_next/image?url=http%3A%2F%2Fi.annihil.us%2Fu%2Fprod%2Fmarvel%2Fi%2Fmg%2Fc%2Fe0%2F535fecbbb9784.jpg&w=384&q=75"
        srcset="/_next/image?url=http%3A%2F%2Fi.annihil.us%2Fu%2Fprod%2Fmarvel%2Fi%2Fmg%2Fc%2Fe0%2F535fecbbb9784.jpg&w=256&q=75 1x, /_next/image?url=http%3A%2F%2Fi.annihil.us%2Fu%2Fprod%2Fmarvel%2Fi%2Fmg%2Fc%2Fe0%2F535fecbbb9784.jpg&w=384&q=75 2x"
        style="color: transparent;"
        width="192"
      />
    </div>
    <div
      class="flex flex-col justify-center items-center w-full h-36"
    >
      <span
        class="text-white text-center"
      >
        Koishi Komeiji
      </span>
      <span
        class="text-white text-center"
      />
    </div>
  </a>
</li>
`;
